src/packages/auth/adapters/CAMAuthAdapter.ts

```diff
@@ -0,0 +1,114 @@
+import { getEnv } from '../../../env.js';
+import { generateJwt, getUserRoles } from "../functions.js";
+import type { AuthAdapter, AuthResponse, ValidateResponse } from "../types.js";
+
+type CAMValidateResponse = {
+  validated?: boolean;
+  errorCode?: string;
+  errorMessage?: string;
+};
+
+type CAMInvalidateResponse = {
+  invalidated?: boolean;
+  errorCode?: string;
+  errorMessage?: string;
+};
+
+type CAMLoginResponse = {
+  userId?: string;
+  errorCode?: string;
+  errorMessage?: string;
+};
+
+export const CAMAuthAdapter: AuthAdapter = {
+
+  logout: async (cookies: any): Promise<boolean> => {
+
+    const { AUTH_SSO_TOKEN_NAME, AUTH_URL } = getEnv();
+
+    const ssoToken = cookies[AUTH_SSO_TOKEN_NAME];
+
+    const body = JSON.stringify({ ssoToken });
+    const url = `${AUTH_URL}/ssoToken?action=invalidate`;
+    const response = await fetch(url, { body, method: 'DELETE' });
+    const { invalidated = false } = await response.json() as CAMInvalidateResponse;
+
+    return invalidated;
+  },
+
+  validate: async (cookies: any): Promise<ValidateResponse> => {
```

Comment:
I suggest just taking the request as the parameter to these functions. You could then get the return URL programmatically via [`req.originalUrl`](https://expressjs.com/en/api.html#req.originalUrl)